spring-cloud-stream-binder-rabbit-core/src/main/java/org/springframework/cloud/stream/binder/rabbit/provisioning/RabbitExchangeQueueProvisioner.java

```diff
@@ -148,8 +148,11 @@ public ConsumerDestination provisionConsumerDestination(String name, String grou
 	private ConsumerDestination doProvisionConsumerDestination(String name, String group,
 			ExtendedConsumerProperties<RabbitConsumerProperties> properties) {
 		boolean anonymous = !StringUtils.hasText(group);
-		String  baseQueueName = anonymous ? groupedName(name, ANONYMOUS_GROUP_NAME_GENERATOR.generateName())
-					: properties.getExtension().isQueueNameGroupOnly() ? group : groupedName(name, group);
+		String  baseQueueName = properties.getExtension().isQueueNameGroupOnly()
+				? anonymous ? ANONYMOUS_GROUP_NAME_GENERATOR.generateName()
+							: group
+				: anonymous ? groupedName(name, ANONYMOUS_GROUP_NAME_GENERATOR.generateName())
+							: groupedName(name, group);
```

Comment:
I find this nested ternary operator usage as too complicated.
I'm not sure that something has not been missed as it happened before with also not so simple ternary operator...